geonode/inspire/metadata.py

```diff
@@ -0,0 +1,306 @@
+import logging
+from rest_framework.reverse import reverse
+
+from django.db.models import Q
+from django.utils.translation import gettext as _
+
+from geonode.base.models import ResourceBase, RestrictionCodeType, ThesaurusKeywordLabel
+from geonode.metadata.handlers.abstract import MetadataHandler
+from geonode.metadata.handlers.sparse import sparse_field_registry
+from geonode.metadata.manager import metadata_manager
+from geonode.metadata.handlers.thesaurus import TKeywordsHandler
+from geonode.metadata.models import SparseField
+
+logger = logging.getLogger(__name__)
+
+
+FIELD_RESTRICTION_TYPE = "restriction_code_type"
+FIELD_RESTRICTION_PUBLIC_ACCESS = "constraints_other"
+FIELD_RESTRICTION_ACCESS_USE = "access_and_use"
+
+THESAURUS_PUBLIC_ACCESS = "limitationsonpublicaccess"
+THESAURUS_ACCESS_USE = "conditionsapplyingtoaccessanduse"
+
+FIELDVAL_ACCESS_USE_FREETEXT = "http://inspire.ec.europa.eu/metadata-codelist/ConditionsApplyingToAccessAndUse/freeText"
+# Should be "otherRestrictions", see https://github.com/GeoNode/geonode/issues/12745
+FIELDVAL_RESTRICTION_TYPE_DEFAULT = "limitation not listed"
+
+CONTEXT_ID = "inspire"
+
+
+class INSPIREHandler(MetadataHandler):
```

Comment:
From my undestanding, the inspire app is not totally required, is more a documentation app. If is not used, i would propose to switch it as documentation or if is needed for testing purpose only, i would document this and move the app in the tests.
Is open to discussion

Reply:
Pls note that I used the inspire app for an example in this very thread (https://github.com/GeoNode/geonode/pull/12794#issuecomment-2586817938). It is a really good point for documentation and testing handler overriding. I know it is something not used outside Europe, but I guess we need some real override implementation inside geonode.